### PR TITLE
tcp: Make firmware strings (in packet suffix) optional

### PR DESCRIPTION
**Pull request by MarijnS95**
Fixes https://github.com/robbinjanssen/home-assistant-omnik-inverter/issues/142, https://github.com/robbinjanssen/home-assistant-omnik-inverter/discussions/134#discussioncomment-5601610

Multiple of the following reports are going around:

    Buffer size too small (85 instead of at least 125 bytes)

As it turns out these last 40 bytes contain exactly the `firmware` and `firmware_slave` strings, which are not sent by certain 3000TL inverters.  Move the fields to a secondary, optional `Structure` to allow these devices to be read out over TCP, and warn when the length of a reply isn't matching one of the known message sizes so that we can potentially discover more different layouts.

### CODE REVIEW

File: omnikinverter/tcp.py [hunk 1]
```diff
@@ -70,12 +70,100 @@ class _TcpData(BigEndianStructure):
         ("padding1", c_ubyte * 4),
         ("unknown0", c_ushort),
         ("padding2", c_ubyte * 10),
+    ]
+
+    @classmethod
+    def parse(cls: _TcpData, data: bytes, offset: int = 0) -> dict[str, Any]:
```
Comment (MarijnS95) [inline omnikinverter/tcp.py:76]: `mypy` and `ruff` are really fighting over `classmethod`s: unsure how to solve their warnings or if this pattern is even desired.

File: tests/test_tcp_models.py [hunk 2]
```diff
@@ -297,6 +297,44 @@ async def test_inverter_tcp() -> None:
...
+
+    assert inverter
+    assert inverter.solar_rated_power is None
+    assert inverter.solar_current_power == 0
+
+    assert inverter.model is None
+    assert inverter.serial_number == "NLDN202013212035"
+    assert inverter.temperature == 23.400000000000002
+    assert inverter.dc_input_voltage == [118.30000000000001, 0.0, None]
+    assert inverter.dc_input_current == [0.0, 15.100000000000001, None]
```
Comment (MarijnS95) [inline tests/test_tcp_models.py:323]: Perhaps a data / parser error? Seems odd to have an unconnected string report a positive Amperage, but no Voltage.

Comment (MarijnS95) [reply tests/test_tcp_models.py:323]: @PeeVv I'm using your dump from https://github.com/robbinjanssen/home-assistant-omnik-inverter/issues/142#issuecomment-1507430793 for this test-case if that's okay. If not I recommend you to remove the public link and I'll anonymize this packet :)

Comment (PeeVv) [reply tests/test_tcp_models.py:323]: > @PeeVv I'm using your dump from [robbinjanssen/home-assistant-omnik-inverter#142 (comment)](https://github.com/robbinjanssen/home-assistant-omnik-inverter/issues/142#issuecomment-1507430793) for this test-case if that's okay. If not I recommend you to remove the public link and I'll anonymize this packet :)

Fine by me! You can use it for whatever you want.

File: omnikinverter/tcp.py (hunk 1 above)
Comment (MarijnS95) [reply omnikinverter/tcp.py]: Ah we simply had to allow `ANN102` in addition to `ANN101`.

**Review by klaasnicolaas — APPROVED** [review]
✅ LGTM

**Pull request merged**